### DISCUSSION TITLE
Make separate functions for interpolation of gradient and laplacian of density

### DISCUSSION THREAD

**Discussion by gabrielasd** [theochem/AtomDB#31] [Ideas]
@PaulWAyers, @FarnazH 

Is there a preference to have the evaluation of the gradient and laplacian of the density linked to the same function that computes the density rather than separate ones?
Currently, for an atomic species, the density (dens), and its gradient and laplacian in the radial direction (d_dens and d2_dens, respectively) are evaluated through the function `interpolate_dens` which is used as follows:
https://github.com/theochem/AtomDB/blob/1830ecff53b7bcef9b47e6635d4fe8696750a083/atomdb/api.py#L212-L217

I don't clearly remember, but I think one reason we made it like this was to homogenize across all datasets the way the gradient/laplacian would get evaluated. And this happened because for datasets using gaussian basis functions we didn't have other way to get spherically averaged gradient/laplacian if not through the spline of the density.
But @marco-2023 has figured a way to directly evaluate these properties in these cases, so now instead of only storing the density array, we could store the three properties for all ab-initio source data. (This would also solve the pending issue that we do not have good evaluation of the laplacian through the 2nd derivative of the density spline)

If we made this change, we would have to modify `interpolate_dens` so that the user can access the data. And talking 
to @marco-2023 and @msricher we though one option would be to add independent interpolation methods for each property.

Could I have your opinion?

**Comment by msricher** [comment]
> we though one option would be to add independent interpolation methods for each property.

I think this is the best idea.

**Reply by PaulWAyers** [reply]
I agree with @msricher . I think it is more straightforward and also more accurate.
